cpp/src/slice2py/PythonUtil.cpp

```diff
@@ -186,29 +186,77 @@ namespace
     }
 
     /// Returns a DocString formatted link to the provided Slice identifier.
-    /// TODO: this is temporary and will be replaced when we add 'python:identifier' support.
-    string pyLinkFormatter(const string& rawLink, const ContainedPtr&, const SyntaxTreeBasePtr&)
+    string pyLinkFormatter(const string& rawLink, const ContainedPtr&, const SyntaxTreeBasePtr& target)
     {
-        ostringstream os;
-        os << "`";
-
-        auto hashPos = rawLink.find('#');
-        if (hashPos != string::npos)
+        ostringstream result;
+        if (target)
         {
-            if (hashPos != 0)
+            if (auto builtinTarget = dynamic_pointer_cast<Builtin>(target))
             {
-                os << rawLink.substr(0, hashPos);
-                os << ".";
+                if (builtinTarget->kind() == Builtin::KindObject)
+                {
+                    result << ":class:`Ice.Object`";
+                }
+                else if (builtinTarget->kind() == Builtin::KindValue)
+                {
+                    result << ":class:`Ice.Value`";
+                }
+                else if (builtinTarget->kind() == Builtin::KindObjectProxy)
+                {
+                    result << ":class:`Ice.ObjectPrx`";
+                }
+                else
+                {
+                    result << "``" << typeToDocstring(builtinTarget, false) << "``";
+                }
+            }
+            else
+            {
+                if (auto operationTarget = dynamic_pointer_cast<Operation>(target))
+                {
+                    string targetScoped = operationTarget->interface()->mappedScoped(".").substr(1);
+
+                    // link to the method on the proxy interface
+                    result << ":meth:`" << targetScoped << "Prx." << operationTarget->mappedName() << "`";
+                }
+                else
```

Comment:
I think this would be better as:
```
if (isBuiltin) {}
elseif (isOperation) {}
else {}
```
instead of the current:
```
if (isBuiltin) {}
else {
    if (isOperation) {}
    else {}
}
```
It's not like we're doing anything else in that `else` block.